TYPO3.Neos/Classes/TYPO3/Neos/Domain/Service/LegacySiteImportService.php

```diff
@@ -456,11 +463,10 @@ protected function importNodeProperty(\SimpleXMLElement $nodePropertyXml, NodeIn
                      * The AssetList node type has a custom implementation to work around this bug.
                      * @see NEOS-121
                      */
-                    $repositoryAction = $this->persistenceManager->isNewObject($entry) ? 'add' : 'update';
-                    if ($entry instanceof Image) {
-                        $this->imageRepository->$repositoryAction($entry);
+                    if ($this->persistenceManager->isNewObject($entry)) {
+                        $this->assetService->update($entry);
                     } else {
-                        $this->assetRepository->$repositoryAction($entry);
+                        $this->assetService->add($entry);
```

Comment:
nope, all repository 'write' actions are done at the service. Just looks like reversal because I reintroduced the if statement and removed the ternary :)